aten/src/ATen/TensorIterator.cpp

```diff
@@ -466,7 +466,7 @@ void TensorIteratorBase::compute_types(const TensorIteratorConfig& config) {
   //   - checks that the common dtype can safely cast to each output, if requested
   //   - creates temporaries for CPU operations, if needed and requested
   common_device_ = common_device;
-  int max_cpu_scalars_on_non_cpu = config.allow_cpu_scalars_ ? 1 : 0;
+  int max_cpu_scalars_on_non_cpu = config.allow_cpu_scalars_ ? config.num_inputs_ - 1 : 0;
```

Comment:
Is there any other change needed down the line to support this?

Reply:
I think max_cpu_scalars_on_non_cpu supports only binary inputs initially.
But, the number of inputs of the gpu_kernel_with_scalars_ternary I wrote is 3.
Exiting code caused errors when executing gpu_kernel_with_scalars_ternary, because of the above constraint (1 or 0). So I made the constraint support the ternary function.

However,
if the tensors wrapping scalars are used, that change might be reverted, I think.